app/models/invoice.rb

```diff
@@ -96,4 +98,14 @@ def calculate_total_amount
   def set_currency
     invoice_items.each{ |i| i.update(currency: currency) }
   end
+
+  def check_bank_account
+    return true if bank_account_id.blank?
+
+    ba = BankAccount.find_by(id: bank_account_id)
+    return true if ba.blank?
```

Comment:
@yunixon just check `return if bank_account.blank?`. Why do you select bank account yourself?